bin.js

```diff
@@ -31,3 +34,12 @@ function getConfig() {
   }
 }
 
+function getRules(set) {
+  var configFileName = set ? ('eslint-config-' + set) : null
```

Comment:
How could we not see this ...
`getConfigForFile()` of the exposed `CLIEngine` object basically only wraps the functionality of the private api's `Config.getConfig()` m(
(I misinterpreted its documentation as _only_ detecting the applicable rules/config of a particular [JavaScript] file based on its location within the filesystem.)

So, sth like
```JavaScript
var path = require('path')
var eslint = require('eslint')
var rulesCombined = new eslint.CLIEngine().getConfigForFile(path.resolve('../path/to/config.file'))
```

(with `../path/to/config.file` being the package.json or .eslintrc*) indeed seems to just do what we want.

I'll alter my code correspondingly later today or no later than tomorrow.